<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Lorenzo Monticelli" />

<meta name="date" content="2024-11-29" />

<title>GeneRepresentationPackage: A Guide</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">GeneRepresentationPackage: A Guide</h1>
<h4 class="author">Lorenzo Monticelli</h4>
<h4 class="date">2024-11-29</h4>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="co">#&gt; Caricamento del pacchetto richiesto: stats4</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="co">#&gt; Caricamento del pacchetto richiesto: BiocGenerics</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co">#&gt; Caricamento pacchetto: &#39;BiocGenerics&#39;</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="co">#&gt; I seguenti oggetti sono mascherati da &#39;package:stats&#39;:</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="co">#&gt; I seguenti oggetti sono mascherati da &#39;package:base&#39;:</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="co">#&gt;     table, tapply, union, unique, unsplit, which.max, which.min</span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a><span class="co">#&gt; Caricamento del pacchetto richiesto: S4Vectors</span></span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a><span class="co">#&gt; Caricamento pacchetto: &#39;S4Vectors&#39;</span></span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a><span class="co">#&gt; Il seguente oggetto è mascherato da &#39;package:utils&#39;:</span></span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a><span class="co">#&gt;     findMatches</span></span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a><span class="co">#&gt; I seguenti oggetti sono mascherati da &#39;package:base&#39;:</span></span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a><span class="co">#&gt; Caricamento del pacchetto richiesto: IRanges</span></span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-28"><a href="#cb1-28" tabindex="-1"></a><span class="co">#&gt; Caricamento pacchetto: &#39;IRanges&#39;</span></span>
<span id="cb1-29"><a href="#cb1-29" tabindex="-1"></a><span class="co">#&gt; Il seguente oggetto è mascherato da &#39;package:grDevices&#39;:</span></span>
<span id="cb1-30"><a href="#cb1-30" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-31"><a href="#cb1-31" tabindex="-1"></a><span class="co">#&gt;     windows</span></span>
<span id="cb1-32"><a href="#cb1-32" tabindex="-1"></a><span class="co">#&gt; Caricamento del pacchetto richiesto: GenomeInfoDb</span></span>
<span id="cb1-33"><a href="#cb1-33" tabindex="-1"></a><span class="fu">library</span>(GeneRepresentationPackage)</span>
<span id="cb1-34"><a href="#cb1-34" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-35"><a href="#cb1-35" tabindex="-1"></a><span class="co">#&gt; Caricamento pacchetto: &#39;GeneRepresentationPackage&#39;</span></span>
<span id="cb1-36"><a href="#cb1-36" tabindex="-1"></a><span class="co">#&gt; Il seguente oggetto è mascherato da &#39;package:base&#39;:</span></span>
<span id="cb1-37"><a href="#cb1-37" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-38"><a href="#cb1-38" tabindex="-1"></a><span class="co">#&gt;     structure</span></span></code></pre></div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The <code>GeneRepresentationPackage</code> provides a set of S4
classes for protein-coding genes, long non-coding RNAs (lncRNAs), and
microRNAs (miRNAs). These classes are inherited from a virtual
<code>Gene</code> class, which includes common attributes such as ID,
symbol, name, description, and genomic structure.</p>
<hr />
</div>
<div id="creating-gene-objects" class="section level1">
<h1>Creating Gene Objects</h1>
<p>The package includes constructors for creating instances of
<code>ProteinCodingGene</code>, <code>lncRNAGene</code>, and
<code>miRNAGene</code>. These objects include specialized attributes for
each gene type.</p>
</div>
<div id="creating-a-new-protein-coding-gene-object" class="section level1">
<h1>Creating a New Protein-Coding Gene Object</h1>
<p>A <code>ProteinCodingGene</code> object inherits all attributes of
the <code>Gene</code> class, along with <code>proteinID</code> and
<code>proteinSequence</code>.</p>
<div id="example-protein-coding-gene" class="section level3">
<h3>Example: Protein-Coding Gene</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># Define a GRanges object for genomic structure</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>tp53_structure <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>    <span class="at">seqnames =</span> <span class="st">&quot;chr17&quot;</span>,</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>    <span class="at">ranges =</span> <span class="fu">IRanges</span>(<span class="at">start =</span> <span class="dv">7668402</span>, <span class="at">end =</span> <span class="dv">7687550</span>),</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>    <span class="at">strand =</span> <span class="st">&quot;+&quot;</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>)</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co"># Create a ProteinCodingGene object</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>tp53 <span class="ot">&lt;-</span> <span class="fu">createProteinCodingGene</span>(</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>    <span class="at">id =</span> <span class="st">&quot;ENSG000001&quot;</span>,</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>    <span class="at">symbol =</span> <span class="st">&quot;TP53&quot;</span>,</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;Tumor Protein&quot;</span>,</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">&quot;Tumor suppressor gene&quot;</span>,</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>    <span class="at">structure =</span> tp53_structure,</span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>    <span class="at">protein_id =</span> <span class="st">&quot;P04637&quot;</span>,</span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a>    <span class="at">protein_sequence =</span> <span class="st">&quot;MVLSPADKTNVKAAWG&quot;</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a>)</span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a><span class="fu">print</span>(tp53)</span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="co">#&gt; An object of class &quot;ProteinCodingGene&quot;</span></span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="co">#&gt; Slot &quot;protein_id&quot;:</span></span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a><span class="co">#&gt; [1] &quot;P04637&quot;</span></span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a><span class="co">#&gt; Slot &quot;protein_sequence&quot;:</span></span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a><span class="co">#&gt; [1] &quot;MVLSPADKTNVKAAWG&quot;</span></span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-26"><a href="#cb2-26" tabindex="-1"></a><span class="co">#&gt; Slot &quot;id&quot;:</span></span>
<span id="cb2-27"><a href="#cb2-27" tabindex="-1"></a><span class="co">#&gt; [1] &quot;ENSG000001&quot;</span></span>
<span id="cb2-28"><a href="#cb2-28" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-29"><a href="#cb2-29" tabindex="-1"></a><span class="co">#&gt; Slot &quot;symbol&quot;:</span></span>
<span id="cb2-30"><a href="#cb2-30" tabindex="-1"></a><span class="co">#&gt; [1] &quot;TP53&quot;</span></span>
<span id="cb2-31"><a href="#cb2-31" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-32"><a href="#cb2-32" tabindex="-1"></a><span class="co">#&gt; Slot &quot;name&quot;:</span></span>
<span id="cb2-33"><a href="#cb2-33" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tumor Protein&quot;</span></span>
<span id="cb2-34"><a href="#cb2-34" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-35"><a href="#cb2-35" tabindex="-1"></a><span class="co">#&gt; Slot &quot;description&quot;:</span></span>
<span id="cb2-36"><a href="#cb2-36" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tumor suppressor gene&quot;</span></span>
<span id="cb2-37"><a href="#cb2-37" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-38"><a href="#cb2-38" tabindex="-1"></a><span class="co">#&gt; Slot &quot;structure&quot;:</span></span>
<span id="cb2-39"><a href="#cb2-39" tabindex="-1"></a><span class="co">#&gt; GRanges object with 1 range and 0 metadata columns:</span></span>
<span id="cb2-40"><a href="#cb2-40" tabindex="-1"></a><span class="co">#&gt;       seqnames          ranges strand</span></span>
<span id="cb2-41"><a href="#cb2-41" tabindex="-1"></a><span class="co">#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span id="cb2-42"><a href="#cb2-42" tabindex="-1"></a><span class="co">#&gt;   [1]    chr17 7668402-7687550      +</span></span>
<span id="cb2-43"><a href="#cb2-43" tabindex="-1"></a><span class="co">#&gt;   -------</span></span>
<span id="cb2-44"><a href="#cb2-44" tabindex="-1"></a><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<hr />
</div>
</div>
<div id="creating-a-new-long-non-coding-rna-object" class="section level1">
<h1>Creating a New Long Non-Coding RNA Object</h1>
<p>A <code>lncRNAGene</code> object inherits all attributes of the
<code>Gene</code> class, along with <code>lncRNAID</code> and
<code>RNASequence</code>.</p>
<div id="example-long-non-coding-rna-gene" class="section level3">
<h3>Example: Long Non-Coding RNA Gene</h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># Define a GRanges object for genomic structure</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>malat1_structure <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>    <span class="at">seqnames =</span> <span class="st">&quot;chr11&quot;</span>,</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>    <span class="at">ranges =</span> <span class="fu">IRanges</span>(<span class="at">start =</span> <span class="dv">65497758</span>, <span class="at">end =</span> <span class="dv">65503100</span>),</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>    <span class="at">strand =</span> <span class="st">&quot;+&quot;</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>)</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co"># Create a lncRNAGene object</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>malat1 <span class="ot">&lt;-</span> <span class="fu">createLncRNAGene</span>(</span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>    <span class="at">id =</span> <span class="st">&quot;ENSG000002&quot;</span>,</span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>    <span class="at">symbol =</span> <span class="st">&quot;MALAT1&quot;</span>,</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;Metastasis Associated Lung Adenocarcinoma Transcript 1&quot;</span>,</span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">&quot;lncRNA associated with metastasis&quot;</span>,</span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>    <span class="at">structure =</span> malat1_structure,</span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>    <span class="at">lncRNA_id =</span> <span class="st">&quot;NONHSAG000002&quot;</span>,</span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a>    <span class="at">RNA_sequence =</span> <span class="st">&quot;AUGCUACGUGA&quot;</span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a>)</span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a><span class="fu">print</span>(malat1)</span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a><span class="co">#&gt; An object of class &quot;lncRNAGene&quot;</span></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a><span class="co">#&gt; Slot &quot;lncRNA_id&quot;:</span></span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a><span class="co">#&gt; [1] &quot;NONHSAG000002&quot;</span></span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a><span class="co">#&gt; Slot &quot;RNA_sequence&quot;:</span></span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a><span class="co">#&gt; [1] &quot;AUGCUACGUGA&quot;</span></span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a><span class="co">#&gt; Slot &quot;id&quot;:</span></span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a><span class="co">#&gt; [1] &quot;ENSG000002&quot;</span></span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a><span class="co">#&gt; Slot &quot;symbol&quot;:</span></span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a><span class="co">#&gt; [1] &quot;MALAT1&quot;</span></span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-32"><a href="#cb3-32" tabindex="-1"></a><span class="co">#&gt; Slot &quot;name&quot;:</span></span>
<span id="cb3-33"><a href="#cb3-33" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Metastasis Associated Lung Adenocarcinoma Transcript 1&quot;</span></span>
<span id="cb3-34"><a href="#cb3-34" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-35"><a href="#cb3-35" tabindex="-1"></a><span class="co">#&gt; Slot &quot;description&quot;:</span></span>
<span id="cb3-36"><a href="#cb3-36" tabindex="-1"></a><span class="co">#&gt; [1] &quot;lncRNA associated with metastasis&quot;</span></span>
<span id="cb3-37"><a href="#cb3-37" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-38"><a href="#cb3-38" tabindex="-1"></a><span class="co">#&gt; Slot &quot;structure&quot;:</span></span>
<span id="cb3-39"><a href="#cb3-39" tabindex="-1"></a><span class="co">#&gt; GRanges object with 1 range and 0 metadata columns:</span></span>
<span id="cb3-40"><a href="#cb3-40" tabindex="-1"></a><span class="co">#&gt;       seqnames            ranges strand</span></span>
<span id="cb3-41"><a href="#cb3-41" tabindex="-1"></a><span class="co">#&gt;          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span id="cb3-42"><a href="#cb3-42" tabindex="-1"></a><span class="co">#&gt;   [1]    chr11 65497758-65503100      +</span></span>
<span id="cb3-43"><a href="#cb3-43" tabindex="-1"></a><span class="co">#&gt;   -------</span></span>
<span id="cb3-44"><a href="#cb3-44" tabindex="-1"></a><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<hr />
</div>
</div>
<div id="creating-a-new-microrna-object" class="section level1">
<h1>Creating a New MicroRNA Object</h1>
<p>A <code>miRNAGene</code> object inherits all attributes of the
<code>Gene</code> class, along with <code>miRNAID</code> and
<code>seedSequence</code>.</p>
<div id="example-microrna-gene" class="section level3">
<h3>Example: MicroRNA Gene</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co"># Define a GRanges object for genomic structure</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>mir21_structure <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>    <span class="at">seqnames =</span> <span class="st">&quot;chr21&quot;</span>,</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>    <span class="at">ranges =</span> <span class="fu">IRanges</span>(<span class="at">start =</span> <span class="dv">25575833</span>, <span class="at">end =</span> <span class="dv">25575902</span>),</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>    <span class="at">strand =</span> <span class="st">&quot;-&quot;</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>)</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co"># Create a miRNAGene object</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>mir21 <span class="ot">&lt;-</span> <span class="fu">createMiRNAGene</span>(</span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>    <span class="at">id =</span> <span class="st">&quot;ENSG000003&quot;</span>,</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>    <span class="at">symbol =</span> <span class="st">&quot;MIR21&quot;</span>,</span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;microRNA 21&quot;</span>,</span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">&quot;A microRNA associated with cancer&quot;</span>,</span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>    <span class="at">structure =</span> mir21_structure,</span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>    <span class="at">miRNA_id =</span> <span class="st">&quot;MI0000077&quot;</span>,</span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a>    <span class="at">seed_sequence =</span> <span class="st">&quot;AGCUUA&quot;</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a>)</span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a><span class="fu">print</span>(mir21)</span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a><span class="co">#&gt; An object of class &quot;miRNAGene&quot;</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a><span class="co">#&gt; Slot &quot;miRNA_id&quot;:</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a><span class="co">#&gt; [1] &quot;MI0000077&quot;</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a><span class="co">#&gt; Slot &quot;seed_sequence&quot;:</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a><span class="co">#&gt; [1] &quot;AGCUUA&quot;</span></span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a><span class="co">#&gt; Slot &quot;id&quot;:</span></span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a><span class="co">#&gt; [1] &quot;ENSG000003&quot;</span></span>
<span id="cb4-28"><a href="#cb4-28" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-29"><a href="#cb4-29" tabindex="-1"></a><span class="co">#&gt; Slot &quot;symbol&quot;:</span></span>
<span id="cb4-30"><a href="#cb4-30" tabindex="-1"></a><span class="co">#&gt; [1] &quot;MIR21&quot;</span></span>
<span id="cb4-31"><a href="#cb4-31" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-32"><a href="#cb4-32" tabindex="-1"></a><span class="co">#&gt; Slot &quot;name&quot;:</span></span>
<span id="cb4-33"><a href="#cb4-33" tabindex="-1"></a><span class="co">#&gt; [1] &quot;microRNA 21&quot;</span></span>
<span id="cb4-34"><a href="#cb4-34" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-35"><a href="#cb4-35" tabindex="-1"></a><span class="co">#&gt; Slot &quot;description&quot;:</span></span>
<span id="cb4-36"><a href="#cb4-36" tabindex="-1"></a><span class="co">#&gt; [1] &quot;A microRNA associated with cancer&quot;</span></span>
<span id="cb4-37"><a href="#cb4-37" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-38"><a href="#cb4-38" tabindex="-1"></a><span class="co">#&gt; Slot &quot;structure&quot;:</span></span>
<span id="cb4-39"><a href="#cb4-39" tabindex="-1"></a><span class="co">#&gt; GRanges object with 1 range and 0 metadata columns:</span></span>
<span id="cb4-40"><a href="#cb4-40" tabindex="-1"></a><span class="co">#&gt;       seqnames            ranges strand</span></span>
<span id="cb4-41"><a href="#cb4-41" tabindex="-1"></a><span class="co">#&gt;          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span id="cb4-42"><a href="#cb4-42" tabindex="-1"></a><span class="co">#&gt;   [1]    chr21 25575833-25575902      -</span></span>
<span id="cb4-43"><a href="#cb4-43" tabindex="-1"></a><span class="co">#&gt;   -------</span></span>
<span id="cb4-44"><a href="#cb4-44" tabindex="-1"></a><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<hr />
</div>
</div>
<div id="accessor-methods" class="section level1">
<h1>Accessor Methods</h1>
<p>The package provides accessor methods to retrieve or modify
attributes, such as <code>symbol</code>, <code>structure</code>, and
<code>name</code>.</p>
<div id="getting-the-symbol" class="section level2">
<h2>Getting the Symbol:</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">symbol</span>(tp53)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="co">#&gt; [1] &quot;TP53&quot;</span></span></code></pre></div>
</div>
<div id="setting-the-symbol" class="section level2">
<h2>Setting the Symbol:</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">symbol</span>(tp53) <span class="ot">&lt;-</span> <span class="st">&quot;TP53_updated&quot;</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="fu">symbol</span>(tp53)</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co">#&gt; [1] &quot;TP53_updated&quot;</span></span></code></pre></div>
</div>
<div id="accessing-the-genomic-structure" class="section level2">
<h2>Accessing the Genomic Structure:</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">structure</span>(tp53)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="co">#&gt; GRanges object with 1 range and 0 metadata columns:</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="co">#&gt;       seqnames          ranges strand</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co">#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co">#&gt;   [1]    chr17 7668402-7687550      +</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co">#&gt;   -------</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
<div id="setting-a-new-genomic-structure" class="section level2">
<h2>Setting a New Genomic Structure:</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>new_structure <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>    <span class="at">seqnames =</span> <span class="st">&quot;chr1&quot;</span>,</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>    <span class="at">ranges =</span> <span class="fu">IRanges</span>(<span class="at">start =</span> <span class="dv">1000</span>, <span class="at">end =</span> <span class="dv">5000</span>),</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>    <span class="at">strand =</span> <span class="st">&quot;-&quot;</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>)</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="fu">structure</span>(tp53) <span class="ot">&lt;-</span> new_structure</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="fu">structure</span>(tp53)</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="co">#&gt; GRanges object with 1 range and 0 metadata columns:</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="co">#&gt;       seqnames    ranges strand</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a><span class="co">#&gt;   [1]     chr1 1000-5000      -</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a><span class="co">#&gt;   -------</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<hr />
</div>
</div>
<div id="computing-product-length" class="section level1">
<h1>Computing Product Length</h1>
<p>The <code>lengthProduct</code> method calculates the length of a
gene’s product.</p>
<div id="example-with-a-protein-coding-gene" class="section level2">
<h2>Example with a Protein-Coding Gene</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">lengthProduct</span>(tp53)</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="co">#&gt; [1] 16</span></span></code></pre></div>
</div>
<div id="example-with-a-lncrna" class="section level2">
<h2>Example with a lncRNA</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">lengthProduct</span>(malat1)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="co">#&gt; [1] 11</span></span></code></pre></div>
</div>
<div id="example-with-a-microrna" class="section level2">
<h2>Example with a microRNA</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">lengthProduct</span>(mir21)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="co">#&gt; [1] 6</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="co">#&gt; R version 4.4.2 (2024-10-31 ucrt)</span></span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64</span></span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="co">#&gt; Running under: Windows 10 x64 (build 19045)</span></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a><span class="co">#&gt; locale:</span></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a><span class="co">#&gt; [1] LC_COLLATE=Italian_Italy.utf8  LC_CTYPE=Italian_Italy.utf8   </span></span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a><span class="co">#&gt; [3] LC_MONETARY=Italian_Italy.utf8 LC_NUMERIC=C                  </span></span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a><span class="co">#&gt; [5] LC_TIME=Italian_Italy.utf8    </span></span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a><span class="co">#&gt; time zone: Europe/Rome</span></span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a><span class="co">#&gt; tzcode source: internal</span></span>
<span id="cb12-16"><a href="#cb12-16" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb12-17"><a href="#cb12-17" tabindex="-1"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb12-18"><a href="#cb12-18" tabindex="-1"></a><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span id="cb12-19"><a href="#cb12-19" tabindex="-1"></a><span class="co">#&gt; [8] base     </span></span>
<span id="cb12-20"><a href="#cb12-20" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb12-21"><a href="#cb12-21" tabindex="-1"></a><span class="co">#&gt; other attached packages:</span></span>
<span id="cb12-22"><a href="#cb12-22" tabindex="-1"></a><span class="co">#&gt; [1] GeneRepresentationPackage_0.1.0 GenomicRanges_1.58.0           </span></span>
<span id="cb12-23"><a href="#cb12-23" tabindex="-1"></a><span class="co">#&gt; [3] GenomeInfoDb_1.42.0             IRanges_2.40.0                 </span></span>
<span id="cb12-24"><a href="#cb12-24" tabindex="-1"></a><span class="co">#&gt; [5] S4Vectors_0.44.0                BiocGenerics_0.52.0            </span></span>
<span id="cb12-25"><a href="#cb12-25" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb12-26"><a href="#cb12-26" tabindex="-1"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb12-27"><a href="#cb12-27" tabindex="-1"></a><span class="co">#&gt;  [1] httr_1.4.7              cli_3.6.3               knitr_1.49             </span></span>
<span id="cb12-28"><a href="#cb12-28" tabindex="-1"></a><span class="co">#&gt;  [4] rlang_1.1.4             xfun_0.49               UCSC.utils_1.2.0       </span></span>
<span id="cb12-29"><a href="#cb12-29" tabindex="-1"></a><span class="co">#&gt;  [7] jsonlite_1.8.9          htmltools_0.5.8.1       sass_0.4.9             </span></span>
<span id="cb12-30"><a href="#cb12-30" tabindex="-1"></a><span class="co">#&gt; [10] rmarkdown_2.29          evaluate_1.0.1          jquerylib_0.1.4        </span></span>
<span id="cb12-31"><a href="#cb12-31" tabindex="-1"></a><span class="co">#&gt; [13] fastmap_1.2.0           yaml_2.3.10             lifecycle_1.0.4        </span></span>
<span id="cb12-32"><a href="#cb12-32" tabindex="-1"></a><span class="co">#&gt; [16] compiler_4.4.2          XVector_0.46.0          digest_0.6.37          </span></span>
<span id="cb12-33"><a href="#cb12-33" tabindex="-1"></a><span class="co">#&gt; [19] R6_2.5.1                GenomeInfoDbData_1.2.13 bslib_0.8.0            </span></span>
<span id="cb12-34"><a href="#cb12-34" tabindex="-1"></a><span class="co">#&gt; [22] tools_4.4.2             zlibbioc_1.52.0         cachem_1.1.0</span></span></code></pre></div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
